docs/investigation/rbac/README.md

```diff
@@ -0,0 +1,149 @@
+
+## Usage
+
+1. Create `rbac-test` and `rbac-test2` channels in your Slack workspace you'll use for testing.
+1. Set up a Kubernetes cluster, e.g. with `k3d` or `colima`.
+1. Invite Botkube bot to these channels.
+1. Clone this repository and check out the PoC pull request with GH CLI:
+
+   ```bash
+   gh pr checkout 909
+   ```
+
+1. Deploy Botkube with the following values using local chart from the PR:
+
+   ```yaml
+   image:
+     repository: kubeshop/pr/botkube
+     tag: 909-PR
+   communications:
+     "default-group":
+       socketSlack:
+         enabled: true
+         appToken: "xapp-..." # TODO: Set app token
+         botToken: "xoxb-..." # TODO: Set bot token
+         notification:
+           type: "short"
+         channels:
+           "default":
+             name: rbac-test
+             bindings:
+               executors:
+                 - kubectl-read-only
+               sources:
+                 - k8s-err-events
+           "second":
+             name: rbac-test2
+             bindings:
+               executors:
+                 - kubectl-read-only
+               sources:
+                 - k8s-err-events
+   executors:
+     kubectl-read-only:
+       kubectl:
+         enabled: true
+   settings:
+     clusterName: dev
+
+   analytics:
+     disable: true
+   ```
+
+   ```bash
+   helm install botkube --namespace botkube ./helm/botkube --wait --create-namespace -f ~/rbac-poc-values.yaml
+   ```
+
+1. Apply Role and ClusterRole resources:
+
+   ```bash
+   kubectl apply -f ./docs/investigation/rbac/assets
+   ```
+
+1. Test getting pods in the `rbac-test` channel:
+
+   ```
+   @Botkube get po
+   ```
+
+   ```
+   @Botkube get po -n botkube
+   ```
+
+   Do the same in the `rbac-test2` channel. You should get an error:
+
+   ```
+   Error from server (Forbidden): services is forbidden: User "{your-email}" cannot list resource "pods" in API group "" in the namespace "default"
+   exit status 1
+   ```
+
+1. Test getting services on the first channel:
+
+   This will work:
+
+   ```
+   @Botkube get svc
+   ```
+
+   But this command won't:
+
+   ```
+   @Botkube get svc -n botkube
+   ```
+
+   Same with other resources:
+
+   ```
+   @Botkube get ingress
+   ```
+
+1. In the second channel (`rbac-test2`), get all deployments:
+
+   ```
+   @Botkube get deploy -A
+   ```
+
+   This command won't work in the first channel.
+
+You can change permissions for Roles and ClusterRoles in runtime and they will be taken into account on the next command execution.
+
+## Process isolation
+
+We have a full control over how a given plugin is run. That means we can use e.g. `chroot` to isolate the plugin process from the rest of the system, in order to avoid reading sensitive data such as Slack credentials.
+
+I tested it successfully with a simple separate Go app on my machine. However, when I modified the code in Botkube codebase, the plugin's gRPC server exits with an error. 
```

Comment:
Most probably Hashicorp's Go plugin system has system-wide dependency and fails due to chroot.

Reply:
That's true, I also suspect that 👍